src/components/common/SortSelect/SortSelect.jsx

```diff
@@ -0,0 +1,73 @@
+import { useState, useRef, useEffect } from 'react';
+import sortIcon from '@/assets/icons/sort.svg';
+import styles from './SortSelect.module.scss';
+
+const SortSelect = ({ value, onChange, options = [] }) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [isMobile, setIsMobile] = useState(false);
+  const ref = useRef();
+
+  const selectedOption = options.find((opt) => opt.value === value);
+
+  useEffect(() => {
+    const handleResize = () => {
+      setIsMobile(window.innerWidth <= 767);
+    };
+    handleResize();
+    window.addEventListener('resize', handleResize);
+    return () => window.removeEventListener('resize', handleResize);
+  }, []);
+
+  useEffect(() => {
+    const handleClickOutside = (e) => {
+      if (ref.current && !ref.current.contains(e.target)) {
+        setIsOpen(false);
+      }
+    };
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => document.removeEventListener('mousedown', handleClickOutside);
+  }, []);
+
+  return (
+    <div className={styles.wrapper} ref={ref}>
+      {isMobile ? (
+        <button
+          type="button"
+          className={styles.sortIcon}
+          onClick={() => setIsOpen((prev) => !prev)}
+          aria-label="Open sort options"
+        >
+          <img src={sortIcon} alt="Select button" />
+        </button>
+      ) : (
+        <button
+          type="button"
+          className={styles.selectButton}
+          onClick={() => setIsOpen((prev) => !prev)}
+          aria-label="Open sort options"
+        >
+          {selectedOption?.label}
+          <span className={styles.arrow} />
+        </button>
+      )}
+      {isOpen && (
+        <ul className={styles.optionList}>
+          {options.map((opt) => (
+            <li
+              key={opt.value}
+              onClick={() => {
+                onChange(opt.value);
+                setIsOpen(false);
+              }}
+              className={styles.option}
+            >
+              {opt.label}
+            </li>
```

Comment:
❗️ 수정요청
이런 클릭 가능한 요소의 경우 버튼을 사용해주세요. 더 적절한 태그가 있는데 이미지 태그에 onClick 을 주게되면 button에게 기대하는 동작을 제대로 수행하지 못합니다!


```suggestion
            <li key={opt.value} className={styles.option}>
              <button 
                type="button" 
                value={opt.value}
                onClick={() => {
                  onChange(opt.value);
                  setIsOpen(false);
                }}>
                  {opt.label}
                </button>
            </li>
```